src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -658,6 +665,10 @@ class module_elf : public module_impl
           throw std::runtime_error("Invalid symbol name offset " + std::to_string(dynstr_offset));
         auto symname = dynstr->get_data() + dynstr_offset;
 
+        if (!m_scratch_pad_mem_size && (strcmp(symname, Scratch_Pad_Mem_Symbol) == 0)) {
```

Comment:
Since m_scratch_pad_mem_size is initialized as 0, if there are two symbols with "scratch-pad-mem-symbol" in .dynsym section, only the first symbol will be used.